_posts/2020-05-01-release-R119/index.md

```diff
@@ -0,0 +1,109 @@
+---
+layout: post
+title: Snowplow R119 Tycho Magnetic Anomaly Two released with new bad row format generally available
+title-short: Snowplow R119 Tycho Magnetic Anomaly Two
+tags: [snowplow, enrichment, bad-rows, release]
+author: Anton
+category: Releases
+permalink: /blog/2020/05/1/snowplow-release-r119/
+---
+
+We are excited to release [Snowplow R119 Tycho Magnetic Anomaly Two][snowplow-release],
+This release is an important milestone for Snowplow pipeline. It makrs general availability of new bad rows (failed events) format and becomes last umbrella release in history of the project.
+Read on to learn more about Snowplow R119 Tycho Magnetic Anomaly Two,  named after the [powerful Space Odyssey monolithic artefact that breaks itself into smaller objects][tma2].
+
+![tma2-img][tma2]
+
+1. [General availability of bad rows format](#badrows-ga)
+2. [Changes in Snowplow Open Source](#monorepo)     **TODO: FROM ANTON IT FEELS THAT THE HEADING IS UNNECESSARY ALARMING**
+3. [Other changes](#other)
+4. [Upgrading](#upgrading)
+5. [Roadmap](#roadmap)
+6. [Getting help](#help)
+
+<h2 id="badrows-ga">1. General availability of bad rows format</h2>
+
+In [Snowplow R118 Morgantina][r118-post] we introduced a new format for bad rows (failed events), bringing a much more structured approach for events failed at any step of the pipeline.
+We announced R118 as public beta, given the big changes we've made in that release.
+We and our open source community extensively tested all assets from R118 and now we're excited to announce that R119 marks general availability of new bad rows format and all associated functionality.
+
+We identified several bugs in R118 that were related to changes we've made there:
+
+* Events from a POST payload could get lost if the payload contained at least one corrupted event ([#4320][issue-4320])
+* Enrich process could crash with `NullPointerException` in case of empty query parameter in IgluAdapter ([#4330][issue-4320])
+* Enrich process could crash with `NullPointerException` in case of empty query parameter in Snowplow Adapter ([#4324][issue-4324], thanks [Rob Kingston][robkingston] for spotting it!)
+
```

Comment:
Maybe worth mentioning that we took the opportunity to add strong unit testing to the enrichment workflow.

Reply:
That. And maybe MaxMind work @oguzhanunlu did.